test/functional/isVisible.html

```diff
@@ -0,0 +1,87 @@
+<!DOCTYPE html>
+<html lang="en" dir="ltr">
+  <head>
+    <meta charset="utf-8">
+    <link rel="shortcut icon" href="">
+    <title>isVisible functional test</title>
+    <style type="text/css">
+      :root {
+        --size: 100px;
+      }
+      #not-visible-1 {
+        width: 0px;
+        overflow: hidden;
+      }
+      #not-visible-2 {
+        height: 0px;
+        overflow: hidden;
+      }
+      #not-visible-3 {
+        display: none;
+      }
+      #not-visible-4-ancestor {
+        display: none;
+      }
+      #not-visible-5 {
+        opacity: 0;
+      }
+      #not-visible-6-ancestor {
+        opacity: 0;
+      }
+      #not-visible-7-ancestor {
+        width: 0;
+      }
+      #not-visible-7 {
+        overflow: hidden;
+      }
+      #not-visible-8-ancestor {
+        height: 0;
+      }
+      #not-visible-8 {
+        overflow: hidden;
+      }
+      .off-screen {
+        width: var(--size);
+        height: var(--size);
+        border: 1px solid black;
+        position: absolute;
+      }
+      #not-visible-9 {
+        left: calc(-2 * var(--size));
+      }
+      #not-visible-10 {
+        top: calc(-2 * var(--size));
+      }
+      #not-visible-11 {
+        right: calc(100vw + 2 * var(--size));
+      }
+      #not-visible-12 {
+        bottom: calc(100vh + 2 * var(--size));
+      }
+
+    </style>
+  </head>
+  <body>
+    <h1>isVisible functional test</h1>
+    <div id="not-visible-1"></div>
```

Comment:
I think we could reorganize this to make it more comprehensible.

When I first first encounter each tag, I have to look upward to find out what the corresponding style info actually is. Does choosing not to inline the styles actually give us anything? I don't think we are reusing any of the numbered ones. I do like that you factored out `.off-screen`.

Also, it would be great to have some comments, descriptive names, or something for some of the less obvious ones, like the various `calc`s.